src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -82,6 +82,73 @@ GTCodeGen::IntervalDefinitions::IntervalDefinitions(const Stencil& stencil)
                               Interval::makeCodeGenName(stencilFun->getInterval()));
 }
 
+/// @brief The StencilFunctionReader class parses a stencil function that is used as a boundary
+/// condition into it's stringstream. In order to use stencil_functions as boundary conditions, we
+/// need them to be memnbers of the stencil-wrapper class. The goal is to template the function s.t
+/// every field is a template argument.
+class StencilFunctionReader : public ASTStencilDesc {
```

Comment:
What is the reason for inhering from ASTStencilDesc?
It seems they are completely different visitors, and we dont use any code from ASTStencilDesc? 
I was wondering since also we need to pass `StencilIDToStencilNameMap` to this BC generator, which has no semantic in this context, and therefore we pass to ctr a empty map, since it is not used.